l10n_be_mis_reports/script/generate_xml.py

```diff
@@ -0,0 +1,465 @@
+import json
+import logging
+import os
+import re
+from xml.dom import minidom
+
+import click
+
+# CONST
+MIS_REPORT_MODEL = "mis.report"
+MIS_REPORT_KPI_MODEL = "mis.report.kpi"
+BS_REPORT_XMLID = "mis_report_bs"
+PL_REPORT_XMLID = "mis_report_pl"
+SPECIALSTYLES = ["9900", "9901", "9902", "9903", "9904"]
+PATH = "../data/"
+
+logging.basicConfig(level=logging.INFO)
+_logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.option("--author", help="name of the copyright author")
+@click.option("--year", help="date of the copyright")
+@click.argument("data-file", type=click.File())
+@click.argument("calc-file", type=click.File())
+def generate_xml(author, year, data_file, calc_file):
+    """This script generates two xml documents for the balance sheet
+    and the profit and loss table from DATA_FILE and CALC_FILE
+
+    DATA_FILE is a json file where all the data of the report is stored
+
+    CALC_FILE is a json file where accounts' calculations are stored
+    """
+    try:
+        os.mkdir(PATH)
+        _logger.info(f"{PATH} created")
+    except FileExistsError:
+        _logger.info(f"{PATH} already exists")
+
+    try:
+        json_data_file = json.load(data_file)
+        json_calc_file = json.load(calc_file)
+        _generate_balance_sheet(author, year, json_data_file, json_calc_file)
+        _generate_profit_loss(author, year, json_data_file, json_calc_file)
+    except json.decoder.JSONDecodeError:
+        _logger.error("Files must be json files")
+
+
+def _generate_balance_sheet(author, year, data_file, calc_file):
+    """Generates the file 'mis_report_bs.xml'"""
+
+    balance_sheet = data_file["internalModels"][0]["editorModel"]["sectionsOrTables"][2]
+    fact_protorypes = data_file["internalModels"][0]["factPrototypes"]
+
+    impl = minidom.getDOMImplementation()
+    xml_bs_document = impl.createDocument(None, "odoo", None)
+    xml_bs_document = _write_copyright(xml_bs_document, author, year)
+    bs_doc_name = balance_sheet["section"]["labels"]["en"]
+    xml_bs_document = _write_report_record(
+        xml_bs_document, bs_doc_name, BS_REPORT_XMLID
+    )
+    xml_bs_document = _write_report_kpi_records(
+        xml_bs_document, balance_sheet, calc_file, BS_REPORT_XMLID, fact_protorypes
+    )
+
+    xml_bs_document.writexml(
+        open(PATH + "new_mis_report_bs.xml", "w"),
+        addindent="  ",
+        newl="\n",
+    )
+
+    _logger.info("file mis_report_bs.xml sucessfully created.")
```

Comment:
You can improve readability by:
 * writing structured docstring for your methods (here is a reStructuredText docstring example)
 * remove useless whitespaces
 * be sure variables have explicit names

(apply these for other methods)

```suggestion
    def _generate_balance_sheet(author, year, data_dict, calc_dict):
        """
        This method generates the MIS Report and its lines (KPI's) in
        a mis_report_bs.xml file.
        :param year: (string) report fiscal year
        :param data_dict: data dictionary containing ... (TODO: explain)
        :param calc_dict: computation dictionary containing ... (TODO: explain)
        """
    balance_sheet = data_file["internalModels"][0]["editorModel"]["sectionsOrTables"][2]
    fact_protorypes = data_file["internalModels"][0]["factPrototypes"]
    impl = minidom.getDOMImplementation()
    xml_bs_document = impl.createDocument(None, "odoo", None)
    xml_bs_document = _write_copyright(xml_bs_document, author, year)
    bs_doc_name = balance_sheet["section"]["labels"]["en"]
    xml_bs_document = _write_report_record(
        xml_bs_document, bs_doc_name, BS_REPORT_XMLID
    )
    xml_bs_document = _write_report_kpi_records(
        xml_bs_document, balance_sheet, calc_file, BS_REPORT_XMLID, fact_protorypes
    )
    xml_bs_document.writexml(
        open(PATH + "new_mis_report_bs.xml", "w"),
        addindent="  ",
        newl="\n",
    )
    _logger.info("file mis_report_bs.xml sucessfully created.")
```

`open(PATH + "new_mis_report_bs.xml", "w")` looks suspicious. It should be managed through a context manager.